docker/Dockerfile

```diff
@@ -0,0 +1,43 @@
+FROM debian:stable-slim
+
+# silence env warnings
+ENV DEBIAN_FRONTEND noninteractive
+
+RUN apt-get update \
+    && apt-get install -y --no-install-recommends \
+    python3-minimal \
+    python3-pip \
+    python3-setuptools \
+    ca-certificates \
+    nodejs \
+    npm \
+    git 
+
+# NOTE: there is currently a compability warning with the target versions of Node vs. NPM - using latest until something is identified as the target version
+RUN rm -rf /var/lib/apt/lists/* \
+    && npm install npm@latest -g
+
+# NOTE: Python module for pip requires upgrade adhead of PIO install to work around defects
+RUN python3 -m pip install -U pip
```

Comment:
*Nit / rant / OT (?)*

Perhaps I was living in a bubble so far, since I used very little of debian-like stuff outside of desktop / real servers. I guess this works for now, but I just tried this one:
```dockerfile
FROM fedora:latest
RUN dnf -y install nodejs python python-pip && dnf clean all
RUN pip install platformio
```
Since we want bleeding-edge sw anyways... Feels like a waste doing all of this with stable debian :)
(...and gyp also pulls python2, since we need to wait until march-april for the next release including it, or pull from testing repos...)

Reply:
doesn't really matter to me - just happened to be what was there and it looked like the repo needed more than just the python or node images.

i will say that i'd still recommend explicitness to avoid any random breakage (python3 vs. python :) )